tests/test_slurm_command_gen_strategy.py

```diff
@@ -426,25 +426,51 @@ def setup_method(self):
 
     def assert_positional_lines(self, lines: list[str]):
```

Comment:
With this update this function name doesn't make much sense anymore. Can we stick to positional checks? We can also switch to use `_append_sbatch_directives` directly for `sbatch` args.

Reply:
The hardcoding of line number  or exact positioning doesn't make sense. It's not that it's a feature mandated by Slurm directive either.  I can change the function name for this test. Goal is to enable feature and have flexibility. Not to fine tune the logic because of a unit test.

Reply:
Right, Slurm doesn't care about lines ordering, but this will pass the check, yet won't work: `#SBATCH ... #SBATCH ... #SBATCH ...`. So this line hardcoding is to make sure _our_ code is correct.

Reply:
Sure but I think this would still fail. It's actually checking for the whole line after #SBATCH. If you only have` #SBATCH ... #SBATCH ... #SBATCH`  this would fail too? Given we have other unit tests to check for mandatory args. It definitely has to enter one of those conditions and fail that assertion.

In general I don't think the test need to enforce in what order sbatch directives needs to be generated given Slurm doesn't enforce that. Except that `#!/bin/bash` line Slurm doesn't enforce anything. Hardcoding that line means in that order means one has to be aware of the unit test and modify their feature logic accordingly. That's like working backwards. We need to add feature and modify the test if needed. Not the other way around IMO.

Let's step back....the method name and logic are divergent. I agree with you on that. I can fix that. But I don't think hardcoding the line numbers is the right thing to do.